tomcat8/build.gradle

```diff
@@ -24,13 +24,20 @@ dependencies {
 // NB: We should never add these directories using the 'sourceSets' directive because that will make
 //     them added to more than one project and having a source directory with more than one output directory
 //     will confuse IDEs such as IntelliJ IDEA.
-tasks.compileJava.source "${rootProject.projectDir}/tomcat9/src/main/java"
-tasks.compileJava.exclude '**/ConfigFileLoaderInitializer*'
-tasks.processResources.from "${rootProject.projectDir}/tomcat9/src/main/resources"
-tasks.compileTestJava.source "${rootProject.projectDir}/tomcat9/src/test/java"
-tasks.processTestResources.from "${rootProject.projectDir}/tomcat9/src/test/resources"
-tasks.sourcesJar.from "${rootProject.projectDir}/tomcat9/src/main/java"
-tasks.sourcesJar.from "${rootProject.projectDir}/tomcat9/src/main/resources"
-tasks.sourcesJar.exclude '**/ConfigFileLoaderInitializer*'
-tasks.javadoc.source "${rootProject.projectDir}/tomcat9/src/main/java"
-tasks.javadoc.exclude '**/ConfigFileLoaderInitializer*'
+def tomcat9ProjectDir = "${rootProject.projectDir}/tomcat9"
+// Copy common files from tomcat9 module to gen-src directory in order to use them as a source set.
+task generateSources(type: Copy) {
+    from "${tomcat9ProjectDir}/src/main/java"
+    into "${project.ext.genSrcDir}/main/java"
+    exclude '**/ArmeriaProcessor*'
+    exclude '**/ConfigFileLoaderInitializer*'
+    exclude '**/package-info.java'
+}
+
+tasks.compileJava.dependsOn(generateSources)
+tasks.compileTestJava.source "${tomcat9ProjectDir}/src/test/java"
+tasks.processResources.from "${tomcat9ProjectDir}/src/main/resources"
+tasks.processTestResources.from "${tomcat9ProjectDir}/src/test/resources"
+tasks.sourcesJar.from "${tomcat9ProjectDir}/src/main/resources"
+
+tasks.compileTestJava.dependsOn(project(':tomcat9').tasks.compileTestJava)
```

Comment:
nit: no new line at the end of the file.

Reply:
Do we need `tasks.compileTestJava.dependsOn(project(':tomcat9').tasks.compileTestJava)`?
It seems there's nothing in `compileTestJava` of `:tomcat9`. 🤔

Reply:
Ah, that's right. I will delete that line.